plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/delete/IcebergPositionDeletePageSink.java

```diff
@@ -0,0 +1,189 @@
+        // which may be present for successfully completing query in presence of failure recovery mechanisms.
+        String fileName = fileFormat.toIceberg().addExtension(session.getQueryId() + "-" + randomUUID());
+        this.outputPath = partition
+                .map(partitionData -> locationProvider.newDataLocation(partitionSpec, partitionData, fileName))
+                .orElseGet(() -> locationProvider.newDataLocation(fileName));
+        JobConf jobConf = toJobConf(hdfsEnvironment.getConfiguration(hdfsContext, new Path(outputPath)));
+        this.writer = fileWriterFactory.createPositionDeleteWriter(new Path(outputPath), POSITION_DELETE_SCHEMA, jobConf, session, hdfsContext, fileFormat.toIceberg());
+    }
+
+    @Override
+    public long getCompletedBytes()
+    {
+        return writtenBytes;
+    }
+
+    @Override
+    public long getMemoryUsage()
+    {
+        return memoryUsage;
+    }
+
+    @Override
+    public long getValidationCpuNanos()
+    {
+        return validationCpuNanos;
+    }
+
+    @Override
+    public CompletableFuture<?> appendPage(Page page)
+    {
+        checkArgument(page.getChannelCount() == 1, "IcebergPositionDeletePageSink expected a Page with only one channel, but got " + page.getChannelCount());
+
+        Block[] blocks = new Block[2];
+        blocks[0] = new RunLengthEncodedBlock(nativeValueToBlock(VARCHAR, utf8Slice(dataFilePath)), page.getPositionCount());
+        blocks[1] = page.getBlock(0);
+        doAppend(new Page(blocks));
+
+        writtenData = true;
+        return NOT_BLOCKED;
+    }
+
+    @Override
+    public CompletableFuture<Collection<Slice>> finish()
+    {
+        Collection<Slice> commitTasks = new ArrayList<>();
+        if (writtenData) {
+            writer.commit();
+            CommitTaskData task = new CommitTaskData(
+                    outputPath,
+                    fileFormat,
+                    writer.getWrittenBytes(),
+                    new MetricsWrapper(writer.getMetrics()),
+                    PartitionSpecParser.toJson(partitionSpec),
+                    partition.map(PartitionData::toJson),
+                    FileContent.POSITION_DELETES,
+                    Optional.of(dataFilePath));
+            Long recordCount = task.getMetrics().recordCount();
+            if (recordCount != null && recordCount > 0) {
+                commitTasks.add(wrappedBuffer(jsonCodec.toJsonBytes(task)));
+            }
+            writtenBytes = writer.getWrittenBytes();
+            validationCpuNanos = writer.getValidationCpuNanos();
+        }
+        else {
+            // clean up the empty delete file
+            writer.rollback();
+        }
+        return completedFuture(commitTasks);
+    }
+
+    @Override
+    public void abort()
+    {
+        writer.rollback();
+    }
+
+    private void doAppend(Page page)
+    {
+        while (page.getPositionCount() > MAX_PAGE_POSITIONS) {
```

Comment:
`_PER_WRITE` or something?

btw this is copied from `io.trino.plugin.iceberg.IcebergPageSink#MAX_PAGE_POSITIONS` but same default is not appropriate here, is it?
RLE of path + block of long is a very narrow "table" compared to generic expectations of the other class.

Why not send the whole `page` to the `writer` at once?
What's the benefit of chunking it?